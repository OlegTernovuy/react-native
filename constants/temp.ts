const OCCUPATION_TYPE = ['Development', 'Testing', 'Engineering'];

const mockData = [
    {
        name: 'Raquel Powlowski',
        avatar: 'https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/672.jpg',
        occupation: 'Development',
        responsibilitiesAtWork:
            'Andy shoes are designed to keeping in mind durability as well as trends, the most stylish range of shoes & sandals',
        id: '1',
    },
];

export { OCCUPATION_TYPE, mockData };
