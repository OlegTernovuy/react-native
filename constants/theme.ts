export const THEME = {
    Colors: {
        primary: '#312651',
        secondary: '#444262',

        lightGray: '#f1f1f1cc',
        lightBlue: '#b1d0ffcc',

        lightWhite: '#ffffff',
    },
    Sizes: {
        small: '4px',
        medium: '8px',
        large: '16px',
        xLarge: '20px',
        xxLarge: '24px',
    },
};
