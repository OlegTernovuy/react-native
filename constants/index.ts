import { OCCUPATION_TYPE } from './temp';
import { THEME } from './theme';

export { OCCUPATION_TYPE, THEME };
