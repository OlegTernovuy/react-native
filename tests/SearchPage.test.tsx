import { fireEvent, render } from '@testing-library/react-native';
import { waitFor } from '@testing-library/react-native';
import MockAdapter from 'axios-mock-adapter';
import { router } from 'expo-router';
import axios from 'axios';

import { mockData } from '../constants/temp';
import SearchPage from '../app/search/[id]';

const mock = new MockAdapter(axios);

jest.mock('expo-router', () => ({
    useGlobalSearchParams: jest.fn().mockReturnValue({ id: 'Raq' }),
    router: {
        push: jest.fn(),
    },
}));

describe('CardDetaild component', () => {
    afterEach(() => {
        mock.reset();
    });

    it('should matches snapshot when data is loaded', async () => {
        mock.onGet('', { params: { name: 'Raq' } }).reply(200, mockData);
        const { toJSON } = render(<SearchPage />);
        await waitFor(() => {
            expect(toJSON()).toMatchSnapshot();
        });
    });

    it('should show loader indicator when data is fetching', async () => {
        mock.onGet('', { params: { name: 'Raq' } }).reply(200, mockData);
        const { getByTestId } = render(<SearchPage />);
        expect(getByTestId('activity-indicator')).toBeTruthy();
    });

    it('should contains cards from search', async () => {
        mock.onGet('', { params: { name: 'Raq' } }).reply(200, mockData);
        const { getByText } = render(<SearchPage />);
        await waitFor(() => {
            mockData.forEach((item) => {
                expect(getByText(item.name)).toBeTruthy();
            });
        });
    });

    it('should navigate to card details page', async () => {
        mock.onGet('', { params: { name: 'Raq' } }).reply(200, mockData);
        const { getByTestId } = render(<SearchPage />);
        const card = await waitFor(() => getByTestId('card-details'));

        fireEvent.press(card);

        expect(router.push).toHaveBeenCalledWith(`/details/1`);
    });

    it('should call router.back on button click', async () => {
        mock.onGet('', { params: { name: 'Raq' } }).reply(200, mockData);
        const { getByTestId } = render(<SearchPage />);
        fireEvent.press(getByTestId('back-button'));
        expect(router.back).toHaveBeenCalled();
    });
});
