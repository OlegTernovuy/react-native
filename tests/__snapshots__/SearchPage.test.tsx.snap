// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetaild component should matches snapshot when data is loaded 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#ffffff",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingBottom": 0,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 0,
    }
  }
>
  <ActivityIndicator
    testID="activity-indicator"
  />
</RCTSafeAreaView>
`;
